homeassistant/components/mqtt/discovery.py

```diff
@@ -28,6 +28,8 @@
     'switch': ['mqtt'],
 }
 
+ALREADY_DISCOVERED = set()
```

Comment:
Don't use globals. Store it as a key in `hass.data` instead, a dictionary for this purpose.